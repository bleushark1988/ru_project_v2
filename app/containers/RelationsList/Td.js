import styled, { css } from 'styled-components';

const Td = styled.td`
  padding: 0.78571429em 0.78571429em;
  text-align: inherit;
  border-top: 1px solid rgba(34, 36, 38, 0.1);
  border-left: 1px solid rgba(34, 36, 38, 0.1);
  &:first-child {
    border-left: none;
  }
  ${props =>
    props.disabled &&
    css`
      color: rgba(40, 40, 40, 0.3);
    `};
`;

export default Td;
