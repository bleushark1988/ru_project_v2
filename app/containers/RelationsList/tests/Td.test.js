import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Td from '../Td';

describe('<Td />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<Td />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should be able to render disabled style', () => {
    const renderedComponent = renderer.create(<Td disabled />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should have a className attribute', () => {
    const renderedComponent = shallow(<Td />);
    expect(renderedComponent.prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<Td id={id} disabled />);

    expect(renderedComponent.prop('id')).toEqual(id);
    expect(renderedComponent.prop('disabled')).toBeDefined();
  });

  it('should not adopt an invalid attribute', () => {
    const renderedComponent = shallow(<Td attribute="test" />);
    expect(renderedComponent.prop('attribute')).toBeUndefined();
  });
});
