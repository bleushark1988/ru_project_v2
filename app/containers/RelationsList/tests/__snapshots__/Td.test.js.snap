// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Td /> should be able to render disabled style 1`] = `
.c0 {
  padding: 0.78571429em 0.78571429em;
  text-align: inherit;
  border-top: 1px solid rgba(34,36,38,0.1);
  border-left: 1px solid rgba(34,36,38,0.1);
  color: rgba(40,40,40,0.3);
}

.c0:first-child {
  border-left: none;
}

<td
  className="c0"
  disabled={true}
/>
`;

exports[`<Td /> should match the snapshot 1`] = `
.c0 {
  padding: 0.78571429em 0.78571429em;
  text-align: inherit;
  border-top: 1px solid rgba(34,36,38,0.1);
  border-left: 1px solid rgba(34,36,38,0.1);
}

.c0:first-child {
  border-left: none;
}

<td
  className="c0"
/>
`;
